### PR TITLE
MEN-4360: Extend the D-Bus API client to fetch the server URL

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
Depends on https://github.com/mendersoftware/mender/pull/701

**Comment by tranchitella** [comment]
Manual pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/243218869

**Review by alfrunes — COMMENTED** [review]
Can we also remove the https configuration options from the `mender-connect` configuration as well?
At least the `Servers` and `ServerURL` options.

**Comment by tranchitella** [comment]
> Can we also remove the https configuration options from the `mender-connect` configuration as well?
> At least the `Servers` and `ServerURL` options.

Servers and ServerURL can go away, but the SSL/HTTPS configuration cannot, as we do not pass those informations over D-BUS. Generally speaking, the config file works as "default", right now, and it is overridden with the value coming up from D-Bus. Not the best design ever, but somehow it makes sense. Any suggestion on how to improve it?

File: app/daemon.go [hunk 1]
```diff
@@ -214,6 +214,10 @@ func (d *MenderShellDaemon) waitForJWTToken(client mender.AuthClient) (string, e
 	for {
 		select {
 		case p := <-tokenStateChange:
+			fmt.Println(p)
+			if len(p) > 1 && p[1].ParamType == dbus.GDBusTypeString && len(p[1].ParamData.(string)) > 0 {
+				d.serverUrl = p[1].ParamData.(string)
+			}
 			if len(p) > 0 && p[0].ParamType == dbus.GDBusTypeString && len(p[0].ParamData.(string)) > 0 {
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply app/daemon.go:221]: Even more: shouldn't we make it fail with `len == 1`? We are changing the unreleased API, so is not that we need to be backwards compatible in any manner.

File: client/dbus/mocks/DBusCallResponse.go [hunk 2]
```diff
@@ -12,8 +12,6 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-// Code generated by mockery v2.1.0. DO NOT EDIT.
-
```
Comment (lluiscampos) [inline client/dbus/mocks/DBusCallResponse.go:16]: Have we edited this file manually? We'll run into conflicts if we want to use `mockery` at a later point.

**Review by lluiscampos — APPROVED** [review]

File: app/daemon.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tranchitella) [reply app/daemon.go:221]: Nope, we need both of them, not only one or the other.

File: app/daemon.go [hunk 3]
```diff
@@ -214,6 +214,10 @@ func (d *MenderShellDaemon) waitForJWTToken(client mender.AuthClient) (string, e
 	for {
 		select {
 		case p := <-tokenStateChange:
+			fmt.Println(p)
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply app/daemon.go:217]: argh, sorry, yes.

File: client/dbus/mocks/DBusCallResponse.go (hunk 2 above)
Comment (tranchitella) [reply client/dbus/mocks/DBusCallResponse.go]: No, I have a different version of mockery and I have to restore the right comments, sorry.
Fixed.

File: app/daemon.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lluiscampos) [reply app/daemon.go:220]: Why? Can you help me understand?

[the remark this answers is not in the document]
Comment (tranchitella) [reply app/daemon.go:220]: @lluiscampos you receive a signal with two strings (params) in it: the first one is the token (handled by the second if), the second one is the server URL (handled by the first if). You need to process both parameters (token, server), thus you'll enter both "if" conditions.

**Pull request merged**